articles/mysql-query-performance-optimization-tips/mysql-query-performance-optimization-tips.md

```diff
@@ -0,0 +1,129 @@
+### Introduction
+
+This article explains common tips for optimizing MySQL queries. As data volume in your database grows, retrieving data from the database and other database operations become complex. This requires more computing resources. Most applications are database driven. Poorly designed SQL queries can significantly downgrade the performance of database-driven applications. According to [MySQL developers guide](https://dev.mysql.com/doc/refman/8.0/en/optimize-overview.html), you can be proactive and plan for optimizations or troubleshoot queries and configurations after experiencing problems. The article will focus on the optimization of individual SQL statements and database structure.
+
+### Prerequisites
+
+To better understand this article, prior knowledge of the MySQL database is essential. Understanding of different SQL queries and how they work is critical. [MySQL Tutorial](https://www.mysqltutorial.org/) and [Tutorials Point](https://www.tutorialspoint.com/mysql/index.htm) have free MySQL tutorials for beginners.
+
+### Benefits of MySQL Database Queries Optimization
+
+We optimize for speed and resources. Optimized queries can run faster and need less computing power.
+
+### Tips for MySQL Queries Optimization
+
+Fast queries are about response time. The goal is to have queries return the required result in the shortest time possible. How much time does a query take to execute? Most of the tools used to query MySQL database give details on time taken to run a query. The most straightforward query cost metrics used in MySQL are query response time, the number of rows scanned, and the number of rows returned. The more the number of rows read, the higher the cost of the query. The screenshots below show the time taken to run a query in both CLI and MySQL Workbench.
+
+![mysql-command-line-interface-query-time-screenshot](mysql-cli-query-time-screenshot.jpg)
+
+![mysql-workbench-query-time-screenshot](mysql-workbench-query-time-screenshot.jpg)
+
+#### Optimizing Queries with EXPLAIN
+
+The EXPLAIN statement provides information about how MySQL executes a statement. EXPLAIN works alongside SELECT, DELETE, INSERT, REPLACE, and UPDATE statements. It displays information from a built-in MySQL optimizer regarding the statement execution plan, and the number of rows scanned in each table. Thus we can determine the cost of the query. For more information about EXPLAIN, check [MySQL documentation](https://dev.mysql.com/doc/refman/8.0/en/using-explain.html). The query below shows how EXPLAIN works with the SELECT statement.
```

Comment:
Need to say something to like, "according to the MySQL documentation" before "EXPLAIN works alongside..."